spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/datastore/GcpDatastoreEmulatorAutoConfiguration.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.datastore;
+
+import com.google.cloud.datastore.testing.LocalDatastoreHelper;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.SmartLifecycle;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * If <code>spring.cloud.gcp.datastore.emulator.enabled</code> is set to <code>true</code>
+ * the emulator will be started as a local datastore server using the
+ * {@link com.google.cloud.datastore.testing.LocalDatastoreHelper}.
+ *
+ * @author Lucas Soares
+ *
+ * @since 1.2
+ */
+@Configuration
+@ConditionalOnProperty(value = "spring.cloud.gcp.datastore.emulator.enabled")
+@AutoConfigureBefore(GcpDatastoreAutoConfiguration.class)
+@EnableConfigurationProperties(GcpDatastoreProperties.class)
+@ConditionalOnMissingBean(LocalDatastoreHelper.class)
+public class GcpDatastoreEmulatorAutoConfiguration implements SmartLifecycle {
+	private static final Log LOGGER = LogFactory.getLog(GcpDatastoreEmulatorAutoConfiguration.class);
+
+	private LocalDatastoreHelper helper;
+
+	private boolean running;
+
+	@Bean
+	public LocalDatastoreHelper createDatastoreHelper(
+			GcpDatastoreProperties datastoreProperties) {
+		EmulatorSettings settings = datastoreProperties.getEmulator();
+
+		this.helper = LocalDatastoreHelper.create(settings.getConsistency(),
+				settings.getPort());
+
+		return this.helper;
+	}
+
+	/** Stops the instance of the emulator. */
+	@Override
+	public void stop() {
+		try {
+			if (LOGGER.isDebugEnabled()) {
```

Comment:
No reason to wrap into the `isDebugEnabled` since your debug message is a constant.